docs/typescript/failures.md

```diff
@@ -61,6 +61,53 @@ When a Workflow or Activity fails with an unhandled error, Temporal checks if th
 
 Workflows and Activities may also throw [`ApplicationFailure.nonRetryable`](https://typescript.temporal.io/api/classes/client.applicationfailure#nonretryable-1) to expressly prevent retries.
 
+Propagated Activity and child Workflow failures are considered retryable and will be retried according to the parent Workflow's retry policy.
+
+The expected behavior is:
+
+- Non retryable application failure -> fails the workflow and cannot be retried
+- Retryable application failure -> fails the workflow and can be retried according to the retry policy
+- Other TemporalFailures -> same as retryable application failure
+- Any other error -> fails the workflow task and can be retried
+
+> Note: Before TypeScript SDK v0.17.0, throwing any error in a Workflow would cause the Workflow execution to fail - in other words, all errors were "non-retryable". The semantics of this was corrected in v0.17.
```

Comment:
```suggestion
> Note: Before TypeScript SDK v0.17.0, throwing any error in a Workflow would cause the Workflow execution to fail - in other words, all errors were "retryable". The semantics of this was corrected in v0.17.
```